test/cmdlineTests/optimizer_inliner_dynamic_reference_constructor/output

```diff
@@ -22,11 +22,20 @@ tag_1:
   not(sub(shl(0x40, 0x01), 0x01))
   and
     /* "optimizer_inliner_dynamic_reference_constructor/input.sol":97:98  f */
-  or(tag_0_12, shl(0x20, tag_4))
+  shl(0x20, tag_4)
   sub(shl(0x40, 0x01), 0x01)
     /* "optimizer_inliner_dynamic_reference_constructor/input.sol":93:98  x = f */
+  swap1
+  dup2
+  and
+    /* "optimizer_inliner_dynamic_reference_constructor/input.sol":97:98  f */
+  tag_0_12
+    /* "optimizer_inliner_dynamic_reference_constructor/input.sol":93:98  x = f */
+  swap1
+  swap2
```

Comment:
Here the value of `A` is `shl(0x20, tag_4)`. Not sure if it should be treated as a constant.

Reply:
No, the tag has to be preserved.

Reply:
I mean the pattern matching chose `A` to be `shl(0x20, tag_4)`. I'm not sure if it should match that.

Reply:
Ah sorry, I thought you meant it can be computed to a specific number. Let me check.

Reply:
Are you sure it's not ` sub(shl(0x40, 0x01), 0x01)`?

Reply:
Sorry, I was wrong. It is indeed that expression.